roles/cluster_infra/templates/resources.tf.j2

```diff
@@ -233,6 +234,30 @@ resource "openstack_networking_port_v2" "{{ partition.name }}" {
 
 {% endfor %}
 
+#####
+##### Deploy key
+#####
+{% if cluster_ssh_private_key_file is not defined %}
+resource "openstack_compute_keypair_v2" "cluster_keypair" {
+  name = "{{ cluster_name }}-deploy-key"
+}
+{% endif %}
+
+## Make a list of all public keys required
+{%- set cluster_keys = [] -%}
+{%- if cluster_ssh_private_key_file is defined -%}
+{{- cluster_keys.append(cluster_ssh_private_key_file) -}}
+{%- else -%}
+{{- cluster_keys.append('\"${openstack_compute_keypair_v2.cluster_keypair.public_key}\"') -}}
+{%- endif -%}
+{%- if cluster_user_ssh_public_key is defined -%}
+{{- cluster_keys.append(cluster_user_ssh_public_key) -}}
+{%- endif -%}
+{%- if cluster_user_additional_public_keys is defined -%}
+{%- for pubkey in cluster_user_additional_public_keys -%}
+{{- cluster_keys.append(pubkey) -}}
+{%- endfor -%}
+{%- endif -%}
```

Comment:
This is dirty. Maybe make a macro that produces the keys and use it in each place it is needed?

Reply:
horrendous indeed, removed